src/test/java/com/google/api/codegen/metacode/SampleInitCodeTest.java

```diff
@@ -120,58 +136,55 @@ public void testNestedListField() throws Exception {
+    InitCodeNode actualStructure = InitCodeNode.newRoot();
+    FieldStructureParser.parse(actualStructure, fieldSpec);
+
+    assertNodeEqual(actualStructure, expectedStructure);
   }
 
   @Test
   public void testFormattedField() throws Exception {
     String fieldSpec = "name%entity";
 
-    HashMap<String, InitValueConfig> initValueMap = new HashMap<>();
     InitValueConfig initValueConfig = InitValueConfig.create("test-api", null);
-    initValueMap.put("name", initValueConfig);
+    InitCodeNode outerStructure = InitCodeNode.createWithValue("name", initValueConfig);
+
+    InitCodeNode expectedStructure = InitCodeNode.newRoot();
+    expectedStructure.mergeChild(outerStructure);
 
-    InitCodeNode expectedStructure = InitCodeNode.createWithValue("name", initValueConfig);
+    InitCodeNode actualStructure = InitCodeNode.newRoot();
+    FieldStructureParser.parse(
+        actualStructure, fieldSpec, ImmutableMap.of("name", initValueConfig));
 
-    InitCodeNode actualStructure = FieldStructureParser.parse(fieldSpec, initValueMap);
-    Truth.assertThat(checkEquals(actualStructure, expectedStructure)).isTrue();
+    assertNodeEqual(actualStructure, expectedStructure);
   }
 
   @Test
   public void testFormattedFieldWithValues() throws Exception {
     List<String> fieldSpecs =
         Arrays.asList("formatted_field%entity1=test1", "formatted_field%entity2=test2");
 
-    HashMap<String, InitValueConfig> initValueMap = new HashMap<>();
     InitValueConfig initValueConfig = InitValueConfig.create("test-api", null);
-    initValueMap.put("formatted_field", initValueConfig);
-
-    HashMap<String, InitValue> expectedCollectionValues = new HashMap<>();
-    expectedCollectionValues.put("entity1", InitValue.createLiteral("test1"));
-    expectedCollectionValues.put("entity2", InitValue.createLiteral("test2"));
 
     InitCodeContext context =
         getContextBuilder()
             .initFieldConfigStrings(fieldSpecs)
-            .initValueConfigMap(initValueMap)
+            .initValueConfigMap(ImmutableMap.of("formatted_field", initValueConfig))
             .build();
     InitCodeNode actualStructure = InitCodeNode.createTree(context);
-    Truth.assertThat(actualStructure.getChildren().isEmpty()).isFalse();
+    assertThat(actualStructure.getChildren().isEmpty()).isFalse();
     InitCodeNode actualFormattedFieldNode = actualStructure.getChildren().get("formatted_field");
-    Truth.assertThat(actualFormattedFieldNode.getInitValueConfig()).isNotNull();
-    Truth.assertThat(
-            actualFormattedFieldNode.getInitValueConfig().hasFormattingConfigInitialValues())
-        .isTrue();
-    Truth.assertThat(
-            actualFormattedFieldNode
-                .getInitValueConfig()
-                .getResourceNameBindingValues()
-                .equals(expectedCollectionValues))
-        .isTrue();
+    assertThat(actualFormattedFieldNode.getInitValueConfig().getResourceNameBindingValues())
```

Comment:
The previous tests all set up `expectedStructure` and compared it to `FieldStructureParser`. Here we are abandoning the pattern and instead are testing `InitCodeNdoe.createTree()`. Why the change? Could we do this consistently?

Reply:
Previously `FieldStructureParser` parses things into a list. Now we have it update the existing tree. So we now have to compare against a tree than a list.

We are doing this consistently right?

Reply:
ok